geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/view/MergedFeatureSourceView.scala

```
@@ -0,0 +1,172 @@
+import org.opengis.filter.expression.PropertyName
+import org.opengis.filter.sort.SortBy
+import org.opengis.util.ProgressListener
+
+/**
+  * Feature source for merged data store view
+  *
+  * @param ds data store
+  * @param sources delegate feature sources
+  * @param sft simple feature type
+  */
+class MergedFeatureSourceView(ds: MergedDataStoreView, sources: Seq[SimpleFeatureSource], sft: SimpleFeatureType)
+    extends SimpleFeatureSource with LazyLogging {
+
+  lazy private val hints = Collections.unmodifiableSet(Collections.emptySet[Key])
+
+  lazy private val capabilities = new MergedQueryCapabilities(sources.map(_.getQueryCapabilities))
+
+  override def getSchema: SimpleFeatureType = sft
+
+  override def getCount(query: Query): Int = sources.map(_.getCount(query)).sum
+
+  override def getBounds: ReferencedEnvelope = {
+    val bounds = new ReferencedEnvelope(org.locationtech.geomesa.utils.geotools.CRS_EPSG_4326)
+    sources.foreach(s => bounds.expandToInclude(s.getBounds))
+    bounds
+  }
+
+  override def getBounds(query: Query): ReferencedEnvelope = {
+    val bounds = new ReferencedEnvelope(org.locationtech.geomesa.utils.geotools.CRS_EPSG_4326)
+    sources.foreach(s => bounds.expandToInclude(s.getBounds(query)))
+    bounds
+  }
+
+  override def getQueryCapabilities: QueryCapabilities = capabilities
+
+  override def getFeatures: SimpleFeatureCollection = getFeatures(Filter.INCLUDE)
+
+  override def getFeatures(filter: Filter): SimpleFeatureCollection = getFeatures(new Query(sft.getTypeName, filter))
+
+  override def getFeatures(query: Query): SimpleFeatureCollection = new MergedFeatureCollection(query)
+
+  override def getName: Name = getSchema.getName
+
+  override def getDataStore: DataStore = ds
+
+  override def getSupportedHints: java.util.Set[Key] = hints
+
+  override def getInfo: ResourceInfo = new DelegatingResourceInfo(this)
+
+  override def addFeatureListener(listener: FeatureListener): Unit = throw new NotImplementedError()
```

Is this consistent with other implementations?  I'd be tempted for this to be an empty block.

---

Yeah, that's how our main one is: https://github.com/locationtech/geomesa/blob/master/geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/geotools/GeoMesaFeatureSource.scala#L98

seems better to fail fast than accept it and ignore it, no?